app/src/test/java/io/intrepid/skeleton/base/BasePresenterTest.java

```diff
@@ -0,0 +1,85 @@
+package io.intrepid.skeleton.base;
+
+import android.support.annotation.NonNull;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import io.intrepid.skeleton.testutils.PresenterTestBase;
+import io.reactivex.Observable;
+import io.reactivex.disposables.Disposable;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class BasePresenterTest extends PresenterTestBase<BasePresenter<BaseContract.View>> {
+    @Mock
+    private BaseContract.View view;
+
+    @Mock
+    private Runnable onBind;
+
+    @Mock
+    private Runnable onUnbind;
+
+    private Disposable disposable;
+
+    @Before
+    public void setUp() {
+        presenter = new TestPresenter(view, testConfiguration);
+    }
+
+    @Test
+    public void bind_onlyOnce() throws Exception {
+        presenter.bindView(view);
+        presenter.bindView(view);
+        assertEquals(view, presenter.view);
+        verify(onBind, times(1)).run();
+    }
+
+    @Test
+    public void unbind_onlyOnce() throws Exception {
+        presenter.unbindView();
+        verify(onUnbind, never()).run();
```

Comment:
Looking at this again, I'd actually split this into two tests, `presenterWithoutViewDoesNotUnbind()` and `unbind_onlyOnce()`.

Reply:
Yeah that's a good call. I was bouncing around a bit in how I split these up since there's a bit of duplication due to lifecycle necessity, but this can definitely be split in two.